gen/calc.py

```diff
@@ -305,15 +305,14 @@ def validate_dcos_overlay_network(dcos_overlay_network):
             "Incorrect value for vtep_subnet: {}."
             " Only IPv4 values are allowed".format(overlay_network['vtep_subnet'])) from ex
 
-    assert 'vtep_subnet6' in overlay_network.keys(), (
-        'Missing "vtep_subnet6" in overlay configuration {}'.format(overlay_network))
-
-    try:
-        ipaddress.ip_network(overlay_network['vtep_subnet6'])
-    except ValueError as ex:
-        raise AssertionError(
-            "Incorrect value for vtep_subnet6: {}."
-            " Only IPv6 values are allowed".format(overlay_network['vtep_subnet6'])) from ex
+    # IPv6 is optional
```

Comment:
@GoelDeepak this makes sense. The only thing that bothers me if some specifies an IPv6 virtual network and doesn't specify `vtep_subnet6` things will break badly?

Reply:
I don't know how to handle that. Any suggestions?

Reply:
We could walk through all the overlay networks present in `dcos_overlay_network` and look for the `ip6` subnet key. If any of the virtual networks has this key than `vtep_subnet6` becomes required?